sematic/ui/src/components/Shell.tsx

```diff
@@ -1,10 +1,20 @@
 import Box from "@mui/material/Box";
 import { ThemeProvider } from "@mui/material/styles";
-import { Outlet } from "react-router-dom";
+import { useContext } from "react";
+import { Navigate, Outlet } from "react-router-dom";
+import { UserContext } from "../index";
+import { useAppContext } from "../hooks/appHooks";
 import createTheme from "../themes/mira";
 import SideBar from "./SideBar";
 
 export default function Shell() {
+  const { authenticatedEnabled } = useAppContext()
+  const { user } = useContext(UserContext);
+
+  if (authenticatedEnabled && !user) {
```

Comment:
Later, if we plan to support several types of authentication channels, we can redirect and branch from here based on the configuration read from the server side.